### PR TITLE
Update pipelines and implement tests

### PR DESCRIPTION
**Pull request by MKoppJones**
- Added tests and enabled them in the pipelines
- Added mypy to pipeline
- Updated comparison method to pretty print a table. This is **intended** to be a dev feature, only use if while getting set up with SSM.
- Fix some small bugs that arose during tests.
- Add typing
- Update setup for pypi

Still to do
- Allow flask config for connection to AWS. It currently needs to be set at the OS env level.
- Update the README

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -27,10 +27,12 @@ jobs:
...
-    #     pytest tests
-    # - name: Coverage Report
-    #   run:
-    #     pytest --cov=./
+    - name: mypy Check
+      run: |
+        mypy $PWD
+    - name: Unit Tests
+      run:
+        pytest tests
```
Comment (CoburnJoe) [inline .github/workflows/main.yml:35]: Line 35 is redundant - line 38 will run the tests as well as calculate the coverage :)

File: flask_environment_manager/environment_manager.py [hunk 2]
```diff
@@ -21,25 +24,75 @@ def __init__(self, app: Flask, path: str = "", region_name: str = ""):
...
+                "",
+                "",
+                colored(f'Total {colored("NO", "red")}', attrs=["bold"]),
+                colored(str(len(missing_params)), "yellow"),
+                colored(str(len(mismatched_params)), "yellow"),
+            ]
+        )
+
+        cprint("SSM Parameter/Current Environment Comparison", attrs=["bold"])
+        print(table)
```
Comment (CoburnJoe) [inline flask_environment_manager/environment_manager.py:81]: Not sure how well this will translate in AWS land

File: tests/test_config.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    def test_building_value_dict(self, app: Flask) -> None:
+        setup_mock_ssm()
+        env_man = EnvironmentManager(app, "/", "eu-west-2")
+        data = env_man._get_ssm_values()
+        assert data == {
+            "test": "9ce10572-a1a4-422c-9cf1-d4b45ecd93c2",
+            "child": "27698a22-c47c-4457-a6b6-907051b4534a",
+        }
+
+    @pytest.mark.parametrize(
```
Comment (CoburnJoe) [inline tests/test_config.py:36]: Tests look comprehensive 👍

**Review by CoburnJoe — DISMISSED** [review]

File: flask_environment_manager/environment_manager.py [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import os
+from typing import Any, Union
 from flask import Flask
 import boto3
+from beautifultable import BeautifulTable
+from termcolor import colored, cprint
```
Comment (nomad-software) [inline flask_environment_manager/environment_manager.py:6]: Personally I wouldn't have any colouring in the output. ANSI escape codes are not handled well in some circumstances.

File: flask_environment_manager/environment_manager.py (hunk 2 above)
Comment (nomad-software) [reply flask_environment_manager/environment_manager.py:81]: Yeah, the only output must only be to the log.

**Review by nomad-software — CHANGES_REQUESTED** [review]

File: flask_environment_manager/environment_manager.py (hunk 4 above)
Comment (MKoppJones) [reply flask_environment_manager/environment_manager.py]: This has been removed, as discussed on slack.

File: flask_environment_manager/environment_manager.py (hunk 2 above)
Comment (MKoppJones) [reply flask_environment_manager/environment_manager.py]: Colors have been removed and print has been swapped to use the app logger

File: .github/workflows/main.yml (hunk 1 above)
Comment (MKoppJones) [reply .github/workflows/main.yml]: Updated

**Review by CoburnJoe — DISMISSED** [review]

File: tests/test_config.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+        "value_in, value_out",
+        [
+            ("", ""),
+            (True, True),
+            (False, False),
+            ("ON", True),
+            ("OfF", False),
+            ("FaLsE", False),
+        ],
```
Comment (swiftace) [inline tests/test_config.py:161]: It'd be good to include string variants of `("true", True)`.

**Review by swiftace — DISMISSED** [review]

File: setup.py [hunk 6]
```diff
@@ -20,5 +20,5 @@
         "Operating System :: OS Independent",
     ],
     python_requires=">=3.8",
-    install_requires=["flask"],
+    install_requires=["flask", "boto3", "beautifultable", "termcolor"],
```
Comment (CoburnJoe) [inline setup.py:23]: Is `termcolor` still required here?

Comment (MKoppJones) [reply setup.py:23]: Nope!

Will remove

**Review by CoburnJoe — APPROVED** [review]

**Review by swiftace — APPROVED** [review]

**Review by nomad-software — APPROVED** [review]

**Pull request merged**